### PR TITLE
User can stop the stopwatch

### PR DESCRIPTION
**Pull request by blakethomas**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,30 +1,44 @@
...
-var limit = null
-
+var id = null
 var $time = document.querySelector('#time')
 var $start = document.querySelector('#start')
 var $stop = document.querySelector('#stop')
 
-function update(){
+function update() {
   return time = time + 1
```
Comment (ronperris) [inline main.js]: You don't need to return here.

File: main.js [hunk 2]
```diff
@@ -1,30 +1,44 @@
...
 }
 
-function updateDom(){
+function updateDom() {
   time.toString()
   return $time.textContent = time
```
Comment (ronperris) [inline main.js]: You don't need to return here.

File: main.js [hunk 3]
```diff
@@ -1,30 +1,44 @@
...
 }
 
-function removeStart(){
-  if(limit !== null){
-    $start.remove()
+function toggleStart() {
+  $start.classList.toggle('startbutton')
+  return
```
Comment (ronperris) [inline main.js]: You don't need to return here.

File: main.js [hunk 4]
```diff
@@ -1,30 +1,44 @@
...
+  if (id === null) {
+    id = window.setInterval(update, 1000)
+    window.setInterval(updateDom, 16)
   }
+}
+
+function stopTimer() {
+  window.clearInterval(id)
+  id = null
   return
```
Comment (ronperris) [inline main.js]: You don't need to return here.

File: main.js [hunk 5]
```diff
@@ -1,30 +1,44 @@
...
 }
 
-$start.addEventListener('click', function(event){
+$start.addEventListener('click', function(event) {
+  event.preventDefault()
```
Comment (ronperris) [inline main.js]: You don't need to prevent default here.

File: main.js [hunk 6]
```diff
@@ -1,30 +1,44 @@
...
+  startTimer()
+  toggleStart()
+})
+
+$stop.addEventListener('click', function(event) {
   event.preventDefault()
```
Comment (ronperris) [inline main.js]: You don't need to prevent default here.

**Review by ronperris — CHANGES_REQUESTED** [review]

**Comment by blakethomas** [comment]
Sorry for all the extra returns, I wasn't realizing that the anonymous function connected to my event listener was going to return those functions for me.

**Pull request merged**